overlord/devicestate/devicemgr.go

```diff
@@ -462,6 +475,47 @@ func (m *DeviceManager) ensureBootOk() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureInstalled() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if release.OnClassic {
+		return nil
+	}
+
+	if m.ensureInstalledRan {
+		return nil
+	}
+
+	if m.recoveryMode != "install" {
+		return nil
+	}
+
+	var seeded bool
+	err := m.state.Get("seeded", &seeded)
+	if err != nil {
+		return err
+	}
+	if !seeded {
+		return nil
+	}
+
+	if m.changeInFlight("install-device") {
+		return nil
+	}
+
+	m.ensureInstalledRan = true
+
+	tasks := []*state.Task{}
+	createPartitions := m.state.NewTask("create-partitions", i18n.G("Create new partitions"))
```

Comment:
we need to think a bit more what's the best name for this task, it will do a bit more than just installing partitions

Reply:
A few possibilities: `bootstrap-system`, `prepare-first-boot`, `prepare-system` (we already have the prepare-device hook so this might sound too similar), `make-bootable`. Maybe @mvo5 could have some good ideas?